### PR TITLE
Add test for checkMetadata function

### PR DESCRIPTION
**Pull request by kavir1698**
Once this test is merged, I will refactor the function and add checks to address #9

### CODE REVIEW

File: datasetIngestor/checkMetadata.go
Comment (minottic) [inline datasetIngestor/checkMetadata.go:23]: is this getHost ever used? if not, maybe we can remove in the next pr

File: datasetIngestor/export_test.go
Comment (minottic) [inline datasetIngestor/export_test.go:1]: what's this file for? can't we simply call getHost in the test?

File: go.mod
Comment (minottic) [inline go.mod:16]: what's this?

File: datasetIngestor/checkMetadata_test.go
Comment (minottic) [inline datasetIngestor/checkMetadata_test.go:27]: I think we shouldn't connect to dacat for real, but rather fake the call response. If we do this, this test has no dependency on dacat and will work even if dacat will change address or if other RIs want to use the CLI

File: datasetIngestor/checkMetadata.go
Comment (kavir1698) [reply datasetIngestor/checkMetadata.go:23]: It is used. See line 138

File: datasetIngestor/export_test.go
Comment (kavir1698) [reply datasetIngestor/export_test.go:1]: No. only functions whose name start with a capital letter are exported, and we need exported functions to call them in the test file.

File: go.mod
Comment (kavir1698) [reply go.mod:16]: Automatically added as a result of imports

File: datasetIngestor/checkMetadata.go
Comment (minottic) [reply datasetIngestor/checkMetadata.go:23]: my bad, sorry!

File: datasetIngestor/checkMetadata_test.go
Comment (kavir1698) [reply datasetIngestor/checkMetadata_test.go:27]: The problem is the tests fail with not information if I change the server.

Comment (minottic) [reply datasetIngestor/checkMetadata_test.go:27]: could it be that it fails [here](https://github.com/paulscherrerinstitute/scicat-cli/blob/3c7a584def4f406a306a2981611d18c8f081d127/datasetIngestor/checkMetadata.go#L228)? If so, and if there isn't a way to mock it in go, I think we could proceed with this PR and do a refactor after that, using in line 228 the client (which we can mock)

Comment (minottic) [reply datasetIngestor/checkMetadata_test.go:27]: maybe in PR #17

Comment (kavir1698) [reply datasetIngestor/checkMetadata_test.go:27]: > could it be that it fails [here](https://github.com/paulscherrerinstitute/scicat-cli/blob/3c7a584def4f406a306a2981611d18c8f081d127/datasetIngestor/checkMetadata.go#L228)?

I tested that too. I created local urls for all the cases `myurl` would be. It fails without any error message.

But yeah, we either can spend an unknown amount of time here, or just mark this part of the test to be improved in the future. Additionally, the APIServer is not an important part of the `CheckMetadata` function, it is used in `GetProposal`, for which we will have independent tests.

#17 just makes the code cleaner and more robust, the logic of the code remains the same.

Shall I mark this line as a TODO?

Comment (minottic) [reply datasetIngestor/checkMetadata_test.go:27]: I think a TODO is fine, maybe you could open an issue for that, so we don't forget. I am fine as well to do it outside of #17, good point.

Comment (kavir1698) [reply datasetIngestor/checkMetadata_test.go]: Ok, I'll create an issue after this is merged.

**Review by minottic — APPROVED** [review]

**Comment by kavir1698** [comment]
There is one change request from @sbliven that's blocking the merge, but I can't see where that is 🤔

**Comment by minottic** [comment]
Spencer has to approve the changes were made, otherwise it's blocked

**Review by sbliven — APPROVED** [review]
I'm not sure what the change I requested was, but this looks fine. Good tests.

**Pull request merged**
